### PR TITLE
NIFI-12925 disallow TRACE and OPTIONS methods in ListenHTTP

### PR DESCRIPTION
**Pull request by markobean**
# Summary

Updates ListenHTTP to disallow TRACE and OPTIONS requests. Any such requests will be sent a 405 response.

[NIFI-12925](https://issues.apache.org/jira/browse/NIFI-12925)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

Verified via new unit tests and within a running NiFi. Used InvokeHTTP to send OPTIONS and TRACE requests to ListenHTTP. Both requests received "405" responses with the FlowFile going to No Retry relationship.

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/servlets/ListenHTTPServlet.java [hunk 1]
```diff
@@ -157,6 +157,21 @@ protected void doHead(final HttpServletRequest request, final HttpServletRespons
         }
     }
 
+    private void notAllowed(final HttpServletRequest request, final HttpServletResponse response) throws IOException {
+        response.sendError(HttpServletResponse.SC_METHOD_NOT_ALLOWED, "Method Not Allowed");
+    }
```
Comment (dan-s1) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/servlets/ListenHTTPServlet.java:162]:
```suggestion
    }
    
```

**Comment by markobean** [comment]
Please apply this commit to the nifi-1.x branch as well.

**Comment by dan-s1** [comment]
@markobean Changes should also be made to `ListenHttp` documentation on lines 84-85 to indicate what TRACE and OPTIONS will return.

**Comment by markobean** [comment]
> @markobean Changes should also be made to `ListenHttp` documentation on lines 84-85 to indicate what TRACE, OPTIONS and CONNECT will return.

Documentation updated.

**Review by exceptionfactory — APPROVED** [review]
Thanks for the improvement @markobean, and thanks for the review @dan-s1, the changes look good! +1 merging

**Comment by markobean** [comment]
Thanks @exceptionfactory for the merge and @dan-s1 for the review.

**Pull request closed without merging**